activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -469,7 +476,12 @@ def find_one(id)
           MSG
         end
 
-        relation = where(primary_key => id)
+        relation = if primary_key.is_a?(Array)
```

Comment:
This one is not using the `klass.composite_primary_key?`

Reply:
Ah, lost it in a rebase. Fixed, thanks!